clients/src/main/java/org/apache/kafka/clients/consumer/Consumer.java

```diff
@@ -116,9 +116,9 @@
     public void seekToEnd(Collection<TopicPartition> partitions);
 
     /**
-     * @see KafkaConsumer#position(TopicPartition)
+     * @see KafkaConsumer#position(TopicPartition, long)
      */
-    public long position(TopicPartition partition);
+    public long position(TopicPartition partition, long timeout);
```

Comment:
Thanks for the review @hachikuji maybe we can avoid the public API change by introducing a new config value, or I am not 100 percent sure about this but maybe we can avoid that timeout thing at all and find a new way to handle this problem. (This needs more investigation from my side:))